hollow-jsonadapter/src/main/java/com/netflix/hollow/jsonadapter/HollowModule.java

```diff
@@ -0,0 +1,268 @@
+                }).map(bean -> {
+                    AnnotatedMethod method = (AnnotatedMethod) bean.getMember();
+                    String methodName = method.getName();
+                    if (methodName.endsWith("Boxed")) {
+                        return bean.rename(new NameTransformer() {
+                            @Override
+                            public String transform(String name) {
+                                int index = name.toLowerCase().indexOf("boxed");
+                                return name.substring(0, index);
+                            }
+
+                            @Override
+                            public String reverse(String transformed) {
+                                return methodName;
+                            }
+                        });
+                    }
+                    return bean;
+                }).collect(Collectors.toList());
+            }
+            return beanProperties;
+        }
+
+        @Override
+        public JsonSerializer<?> modifySerializer(SerializationConfig config, BeanDescription beanDesc, JsonSerializer<?> serializer) {
+            if (beanDesc.getType().isTypeOrSubTypeOf(HollowObject.class)) {
+                List<PropertyWriter> propertyWriters = new ArrayList<>();
+                serializer.properties().forEachRemaining(propertyWriters::add);
+                if (propertyWriters.size() == 1) {
+                    BeanPropertyWriter beanPropertyWriter = (BeanPropertyWriter) propertyWriters.get(0);
+                    if (!beanPropertyWriter.getPropertyType().isAssignableFrom(HollowObject.class)) {
+                        return new SinglePropertyWriterSerializer(beanPropertyWriter);
+                    }
+                }
+            }
+            return serializer;
+        }
+
+        @Override
+        public JsonSerializer<?> modifyKeySerializer(SerializationConfig config, JavaType valueType, BeanDescription beanDesc, JsonSerializer<?> serializer) {
+            if (serializer.handledType() == Object.class) {
+                //noinspection unchecked
+                return new SinglePropertyWriterKeySerializer((JsonSerializer<Object>) serializer);
+            }
+            return serializer;
+        }
+
+        @Override
+        public List<BeanPropertyWriter> orderProperties(SerializationConfig config, BeanDescription beanDesc, List<BeanPropertyWriter> beanProperties) {
+            if (preservePropertyOrder) {
+                List<String> orderedBeanPropertyNames = declaredOrderBeanPropertyNames(beanDesc);
+                beanProperties.sort(Comparator.comparingInt(o -> orderedBeanPropertyNames.indexOf(o.getName())));
+            }
+            return beanProperties;
+        }
+
+        /**
+         * Unfortunately, {@link Class#getDeclaredMethods()} doesn't return methods in their declared order. {@link HollowObject#getSchema()} would allow us to get at the field ordering,
```

Comment:
What is the declared order? There is no guarantee that the order of methods declared in the byte code reflect the order of methods declared in the Java source.